### PR TITLE
Fix cpp SimpleConsumer

### PR DESCRIPTION
**Pull request by yulangz**
Fix the bug that SimpleConsumer in the cpp version cannot obtain the broker Routing table.

### CODE REVIEW

**Comment by lizhanhui** [comment]
Please format your pull-request changes using clang-format

File: cpp/source/client/ClientManagerImpl.cpp [hunk 1]
```diff
@@ -764,7 +764,7 @@ State ClientManagerImpl::state() const {
 }
 
 MessageConstSharedPtr ClientManagerImpl::wrapMessage(const rmq::Message& item) {
-  assert(item.topic().resource_namespace() == resource_namespace_);
+  // assert(item.topic().resource_namespace() == resource_namespace_);
```
Comment (lizhanhui) [inline cpp/source/client/ClientManagerImpl.cpp:767]: Why comment out this assertion?

File: cpp/source/rocketmq/Producer.cpp [hunk 2]
```diff
@@ -87,7 +87,8 @@ ProducerBuilder& ProducerBuilder::withConfiguration(Configuration configuration)
 }
 
 ProducerBuilder& ProducerBuilder::withTopics(const std::vector<std::string>& topics) {
-  impl_->topicsOfInterest(topics);
+  auto topics_copy = topics;
```
Comment (lizhanhui) [inline cpp/source/rocketmq/Producer.cpp:90]: I do not see how the proposed change is superior to the original. Any reason for this change?

Comment (yulangz) [reply cpp/source/rocketmq/Producer.cpp:90]: 1. The input parameter of `ProducerBuilder::withTopics` is a const reference, while the input parameter of `topicsOfInterest` is not a constant reference. The original approach cannot be compiled.

2. The internal implementation of `topicsOfInterest` will use `std:: vector<>. swap()`, which will cause the internal memory space held by the incoming `topics` to be swapped out, which does not comply with the semantics of read-only input parameters for `withTopics`.

File: cpp/source/client/ClientManagerImpl.cpp (hunk 1 above)
Comment (yulangz) [reply cpp/source/client/ClientManagerImpl.cpp:767]: The resource_namespace_ is always "", this causes this assert to fail.

File: cpp/source/rocketmq/Producer.cpp (hunk 2 above)
Comment (lizhanhui) [reply cpp/source/rocketmq/Producer.cpp:90]: > The input parameter of ProducerBuilder::withTopics is a const reference, while the input parameter of topicsOfInterest is not a constant reference. The original approach cannot be compiled.
No, this is not true.  `ProducerImpl::topicsOfInterest(std::vector<std::string> topics)` copy happens implicitly.

> 2. The internal implementation of topicsOfInterest will use std:: vector<>. swap(), which will cause the internal memory space held by the incoming topics to be swapped out, which does not comply with the semantics of read-only input parameters for withTopics.
No, it is using the implicit copy of the topics.

File: cpp/source/client/ClientManagerImpl.cpp (hunk 1 above)
Comment (lizhanhui) [reply cpp/source/client/ClientManagerImpl.cpp:780]: @yulangz It's true `resource_namespace_` is empty for the moment. But the assertion should pass

File: cpp/source/rocketmq/Producer.cpp (hunk 2 above)
Comment (lizhanhui) [reply cpp/source/rocketmq/Producer.cpp:90]: @yulangz

**Review by lizhanhui — CHANGES_REQUESTED** [review]
The pull request understands C++ inaccurate. See this example: https://onecompiler.com/cpp/3zpv8g247

**Pull request closed without merging**